src/parastell.py

```diff
@@ -1,150 +1,647 @@
-        offset (np.array(double)) : for each poloidal angle, an offset in [cm] from the 
-            surface defined by s
-        theta (np.array(double)) : the set of poloidal angles in [rad]
-        phi (double) : the toroidal angle for this rib in [rad]
-        scale (double) : a scaling factor between the units of vmec and [cm] (default: m2cm = 100)
-    '''
-    def __init__(
-            self,
-            vmec_,
-            s_,
-            offset_,
-            theta_,
-            phi_,
-            scale_ = m2cm
-    ):
-        self.vmec = vmec_
-        self.s = s_
-        self.offset = offset_
-        self.theta = theta_
-        self.phi = phi_
-        self.scale = scale_
-
-        if not np.all(self.offset >= 0):
+        self.check_input()
+
+        if (
+            self.export_dict['h5m_export'] == 'cubit' or
+            self.magnets is not None
+        ):
+            self.init_cubit()
+
+    def check_input(self):
+        '''Checks user input for errors.
+        '''
+        if (
+            self.export_dict['h5m_export'] not in
+            [None, 'cubit', 'cad_to_dagmc']
+        ):
             raise ValueError(
-                'Offset must be greater than or equal to 0. Check thickness inputs '
-                'for negative values'
+                'h5m_export must be None or have a string value of \'cubit\' '
+                'or \'cad_to_dagmc\''
             )
 
-        self.r_loci = self.vmec2xyz()
+        if (
+            self.export_dict['h5m_export'] == 'cubit' and
+            not self.export_dict['step_export']
+        ):
+            raise ValueError('H5M export via Cubit requires STEP files')
 
-        if self.offset > 0:
-            self.r_loci += (self.offset.T * self.surf_norm().T).T
+        if (self.export_dict['h5m_export'] == 'cad_to_dagmc' and
+                self.magnets is not None):
+            raise ValueError(
+                'Inclusion of magnets in H5M model requires Cubit export'
+            )
+    
+    def init_cubit(self):
+        '''Initializes Coreform Cubit with the DAGMC plugin.
+        '''
+        cubit_dir = os.path.dirname(inspect.getfile(cubit))
+        cubit_dir = Path(cubit_dir) / Path('plugins')
+        cubit.init([
+            'cubit',
+            '-nojournal',
+            '-nographics',
+            '-information', 'off',
+            '-warning', 'off',
+            '-commandplugindir',
+            str(cubit_dir)
+        ])
 
-        self.r_loci = np.append(self.r_loci, self.r_loci[0,:])
+    def construct_invessel_build(self):
+        '''Construct InVesselBuild class object.
+        '''
+        ivb = InVesselBuild(
+            self.vmec, self.build, self.repeat, self.num_phi, self.num_theta,
+            self.scale, self.export_dict['plasma_h5m_tag'],
+            self.export_dict['sol_h5m_tag'], self.logger
+        )
+        self.invessel_build = ivb.invessel_build
```

Comment:
You import `invessel_build` as `ivb` so I think you are overwriting this variable here.  Instead, let's set it to be the member variable right away. 

```suggestion
        self.invessel_build = InVesselBuild(
            self.vmec, self.build, self.repeat, self.num_phi, self.num_theta,
            self.scale, self.export_dict['plasma_h5m_tag'],
            self.export_dict['sol_h5m_tag'], self.logger
        )
```